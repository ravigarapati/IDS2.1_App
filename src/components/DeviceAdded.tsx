import React from 'react';
import {View, Image, StyleSheet, Dimensions} from 'react-native';
import CustomText from './CustomText';
import Button from './Button';
import FireImage from './../assets/images/deviceadded.svg';

import SUCCESS from '../assets/images/success.svg';

type DeviceAddedProps = {
  header?: string;
  description?: string;
  submit: any;
  deviceType?: string;
  cancelAction?: any;
};

const deviceHeight = Dimensions.get('window').height;

const DeviceAdded = ({
  header,
  description,
  submit,
  deviceType,
  cancelAction,
}: DeviceAddedProps) => {
  return (
    <View style={styles.flex1}>
      <View style={styles.container}>
        {header && (
          <View>
            <CustomText
              allowFontScaling={true}
              style={{
                marginTop: deviceHeight * 0.05,
              }}
              text={header}
              size={21}
              font={'medium'}
            />
          </View>
        )}
        <View
          style={{
            marginTop: deviceHeight * 0.07,
          }}>
          <SUCCESS />
        </View>
        {description && (
          <View>
            <CustomText
              allowFontScaling={true}
              style={{marginTop: deviceHeight * 0.03}}
              font={'regular'}
              text={description}
              size={16}
            />
          </View>
        )}
      </View>
      <View style={styles.buttonSection}>
        <Button
          testID="submitButton"
          type="primary"
          text={'Next'}
          onPress={submit}
        />
        {cancelAction ? (
          <Button
            testID="cancelButton"
            type="secondary"
            text={'Cancel'}
            onPress={cancelAction}
          />
        ) : null}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  flex1: {
    flex: 1,
    backgroundColor: 'white',
  },
  container: {
    flex: 1,
    //justifyContent: 'space-evenly',
    alignItems: 'center',
    marginHorizontal: 40,
  },
  buttonSection: {flex: 1, justifyContent: 'flex-end', marginBottom: 20},
});

export default DeviceAdded;
