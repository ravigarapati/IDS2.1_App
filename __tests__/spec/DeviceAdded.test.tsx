import React from 'react';
import renderer from 'react-test-renderer';
import DeviceAdded from './../../src/components/DeviceAdded';
import {render, cleanup, fireEvent} from 'react-native-testing-library';

describe('Device Added Screen', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(
        <DeviceAdded
          header={' Thermostat'}
          description={
            'Thermostat successfully\nregistered to the user account'
          }
          submit={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const tree = renderer
      .create(
        <DeviceAdded
          header={' Thermostat'}
          description={
            'Thermostat successfully\nregistered to the user account'
          }
          submit={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Cancel button clicked', () => {
    const cancelFuntion = jest.fn();
    const rendered = render(
      <DeviceAdded
        header={' Thermostat'}
        description={'Thermostat successfully\nregistered to the user account'}
        submit={cancelFuntion}
        cancelAction={cancelFuntion}
      />,
    );
    const cancelComponent = rendered.getByTestId('cancelButton');
    fireEvent(cancelComponent, 'press');
    expect(cancelFuntion).toHaveBeenCalled();
  });
  test('Submit button clicked', () => {
    const submitFunction = jest.fn();
    const rendered = render(
      <DeviceAdded
        header={' Thermostat'}
        description={'Thermostat successfully\nregistered to the user account'}
        submit={submitFunction}
      />,
    );
    const submitButton = rendered.getByTestId('submitButton');
    fireEvent(submitButton, 'press');
    expect(submitFunction).toHaveBeenCalled();
  });
});
