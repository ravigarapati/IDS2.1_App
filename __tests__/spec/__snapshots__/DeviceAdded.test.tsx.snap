// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device Added Screen Component to be rendered 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginHorizontal": 40,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 21,
              "lineHeight": 31.5,
            },
            Object {
              "fontFamily": "BoschSans-Medium",
            },
            Object {
              "marginTop": 66.7,
            },
          ]
        }
      >
         Thermostat
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 93.38000000000001,
        }
      }
    >
      <svg
        data-file-name="SvgSuccess"
      />
    </View>
    <View>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
            Object {
              "fontFamily": "BoschSans-Regular",
            },
            Object {
              "marginTop": 40.019999999999996,
            },
          ]
        }
      >
        Thermostat successfully
registered to the user account
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 20,
      }
    }
  >
    <View>
      <View
        accessibilityElementsHidden={false}
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        importantForAccessibility="auto"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginVertical": 5,
              "paddingVertical": 15,
            },
            Object {},
            Object {
              "backgroundColor": "#004975",
            },
            Object {},
            Object {
              "borderColor": "#004975",
              "borderWidth": 1,
            },
          ]
        }
        testID="submitButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Regular",
                },
                Object {},
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
